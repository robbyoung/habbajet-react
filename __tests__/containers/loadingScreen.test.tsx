import React from 'react';
import renderer from 'react-test-renderer';
import {LayoutRoot} from 'react-native-navigation';
import LoadingScreen from '../../app/containers/loadingScreen';
import {Provider} from 'react-redux';
import store from '../../app/store';

let layout: LayoutRoot;
jest.mock('react-native-navigation', () => ({
    Navigation: {
        registerComponent: () => undefined,
        setRoot: (args: LayoutRoot) => (layout = args),
    },
}));

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

let stateLoaded = false;
jest.mock('../../app/storage', () => ({
    loadState: () => {
        stateLoaded = true;
        return {
            purchases: [],
            habbajets: [],
            budget: 0,
            tags: [],
        };
    },
}));

describe('Loading Screen Component', () => {
    it('will redirect to home screen', async () => {
        const component = renderer.create(
            <Provider store={store}>
                <LoadingScreen />
            </Provider>,
        );
        await waitForLoad();

        // @ts-ignore
        expect(layout.root.stack.children[0].component.name).toEqual('Home');
        expect(stateLoaded).toEqual(true);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

async function waitForLoad() {
    return new Promise(resolve => {
        setTimeout(() => resolve(), 0);
    });
}
