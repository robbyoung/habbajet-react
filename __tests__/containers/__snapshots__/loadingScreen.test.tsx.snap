// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Screen Component will redirect to home screen 1`] = `<View />`;
